metecho/api/views.py

```diff
@@ -194,6 +217,18 @@ def get_queryset(self):
         repo_ids = self.request.user.repositories.values_list("repo_id", flat=True)
         return self.queryset.filter(repo_id__in=repo_ids)
 
+    def create(self, request):
+        if not request.user.has_perm("api.add_project"):
+            raise PermissionDenied(_("You are not authorized to create projects"))
```

Comment:
@davisagli I went with Django's regular `add` permission to lock down this endpoint, let me know if you had something else in mind

Reply:
@jerivas I think that's okay as long as this is the only way for a user with this permission but without the `staff` flag to add a project.

Reply:
I'll make sure to test explicitly against that. Should staff users be able to use this endpoint without the permission?

Reply:
Superusers should (I think that's automatic in Django's permission system) but not all staff. My point about staff is that if they have the permission they'll also be able to add projects in the Django admin, which is as expected.

Reply:
Yeah, permission checks always return `True` for superusers. With the current implementation all other staff will need the actual permission just as in the admin.